src/platform_impl/macos/window.rs

```diff
@@ -451,35 +470,50 @@ impl UnownedWindow {
     }
 
     #[inline]
-    pub fn inner_size(&self) -> LogicalSize {
+    pub fn inner_size(&self) -> PhysicalSize {
         let view_frame = unsafe { NSView::frame(*self.ns_view) };
-        (view_frame.size.width as f64, view_frame.size.height as f64).into()
+        let logical: LogicalSize =
+            (view_frame.size.width as f64, view_frame.size.height as f64).into();
+        let dpi_factor = self.hidpi_factor();
+        logical.to_physical(dpi_factor)
     }
 
     #[inline]
-    pub fn outer_size(&self) -> LogicalSize {
+    pub fn outer_size(&self) -> PhysicalSize {
         let view_frame = unsafe { NSWindow::frame(*self.ns_window) };
-        (view_frame.size.width as f64, view_frame.size.height as f64).into()
+        let logical: LogicalSize =
+            (view_frame.size.width as f64, view_frame.size.height as f64).into();
+        let dpi_factor = self.hidpi_factor();
+        logical.to_physical(dpi_factor)
     }
 
     #[inline]
-    pub fn set_inner_size(&self, size: LogicalSize) {
+    pub fn set_inner_size(&self, size: Size) {
         unsafe {
-            util::set_content_size_async(*self.ns_window, size);
+            let dpi_factor = self.hidpi_factor();
+            util::set_content_size_async(*self.ns_window, size.to_logical(dpi_factor));
         }
     }
 
-    pub fn set_min_inner_size(&self, dimensions: Option<LogicalSize>) {
+    pub fn set_min_inner_size(&self, dimensions: Option<Size>) {
         unsafe {
-            let dimensions = dimensions.unwrap_or_else(|| (0, 0).into());
-            set_min_inner_size(*self.ns_window, dimensions);
+            let dimensions = dimensions.unwrap_or(Logical(LogicalSize {
+                width: 0.0,
+                height: 0.0,
+            }));
+            let dpi_factor = self.hidpi_factor();
+            set_min_inner_size(*self.ns_window, dimensions.to_logical(dpi_factor));
         }
     }
 
-    pub fn set_max_inner_size(&self, dimensions: Option<LogicalSize>) {
+    pub fn set_max_inner_size(&self, dimensions: Option<Size>) {
         unsafe {
-            let dimensions = dimensions.unwrap_or_else(|| (!0, !0).into());
-            set_max_inner_size(*self.ns_window, dimensions);
+            let dimensions = dimensions.unwrap_or(Logical(LogicalSize {
+                width: 0.0,
```

Comment:
The documentation for [`maxSize`](https://developer.apple.com/documentation/appkit/nswindow/1419595-maxsize?language=objc) seems to imply that this should be `std::f32::MAX` instead.

Reply:
`LogicalSize` expects dimensions to be `f64`.
Docs also note "that he window server limits window sizes to 10,000."
Maybe we should default to `10000.0`?

Reply:
I think this needs to be `std::f32::MAX`. It's fine that we aren't using the full range of `f64` in `LogicalSize`. The docs mention explicitly that resetting to the default value can only be done by setting this to `FLT_MAX`, which equals `std::f32::MAX`. It might not matter, but it might also be that AppKit is doing an explicit comparison for the default value somewhere, which is why I'd prefer to set this to the exact value. As a sidenote, the data type of `CGFloat` is `f64` on 64-bit operating systems and `f32` on 32-bit operating systems, which is likely why `FLT_MAX` was chosen in AppKit for the default instead of `DBL_MAX`.

Reply:
Ok, I set it tp use f32.